### PR TITLE
doc: net: Enhance IEEE 802.15.4 documentation

### PR DESCRIPTION
**Pull request by jukkar**
Add a simple overview chapter to IEEE 802.15.4 documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
This needs more TLC. We probably need a documentation how the APIs are used.

**Review by tbursztyka — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: doc/reference/networking/ieee802154.rst [hunk 1]
```diff
@@ -5,7 +5,13 @@ IEEE 802.15.4
 
 Overview
 ********
+IEEE 802.15.4 is a technical standard which defines the operation of low-rate
+wireless personal area networks (LR-WPANs). For more detailed overview of this
+standard, see https://en.wikipedia.org/wiki/IEEE_802.15.4
```
Comment (dbkinder) [inline doc/reference/networking/ieee802154.rst]:
```suggestion
standard, see this `IEEE 802.15.4 Wikipedia article <https://en.wikipedia.org/wiki/IEEE_802.15.4>`_.
```

File: doc/reference/networking/ieee802154.rst [hunk 2]
```diff
@@ -5,7 +5,13 @@ IEEE 802.15.4
...
 
+Zephyr supports IEEE 802.15.4 with Thread and 6LoWPAN. The Thread implementation
+is based on https://openthread.io/. The IPv6 header compression in 6LoWPAN is
```
Comment (dbkinder) [inline doc/reference/networking/ieee802154.rst]:
```suggestion
is based on `OpenThread <https://openthread.io/>`_. The IPv6 header compression in 6LoWPAN is
```

**Review by dbkinder — DISMISSED** [review]

**Comment by jukkar** [comment]
Updated according to comments.

**Pull request merged**